cartofacade/capi_test.go

```diff
@@ -113,59 +156,88 @@ func TestCGoAPI(t *testing.T) {
+		t.Log("sensor reading 1")
+		timestamp = timestamp.Add(time.Second * 2)
+		testAddSensorReading(t, vc, "mysensor", "viam-cartographer/mock_lidar/0.pcd", timestamp, pointcloud.PCDAscii)
+
+		// test getPosition zeroed if not enough sensor data has been provided
+		position, err := vc.getPosition()
 		test.That(t, err, test.ShouldBeNil)
+		test.That(t, position.ComponentReference, test.ShouldEqual, "mysensor")
+		positionIsZero(t, position)
 
-		// confirm the pointcloud package still doesn't support binary compressed
-		// pointclouds. If it does, we need to implement:
-		// https://viam.atlassian.net/browse/RSDK-3753
-		err = pointcloud.ToPCD(pc, buf, 2)
+		// test getPointCloudMap returns error if not enough sensor data has been provided
+		pcd, err := vc.getPointCloudMap()
+		test.That(t, pcd, test.ShouldBeNil)
 		test.That(t, err, test.ShouldBeError)
-		test.That(t, err.Error(), test.ShouldResemble, "compressed PCD not yet implemented")
+		test.That(t, err, test.ShouldResemble, errors.New("VIAM_CARTO_POINTCLOUD_MAP_EMPTY"))
 
-		// test getPosition
-		holder, err := vc.getPosition()
+		// 2. test valid addSensorReading: valid reading binary
+		t.Log("sensor reading 2")
+		timestamp = timestamp.Add(time.Second * 2)
+		testAddSensorReading(t, vc, "mysensor", "viam-cartographer/mock_lidar/1.pcd", timestamp, pointcloud.PCDBinary)
 
+		// test getPosition zeroed
+		position, err = vc.getPosition()
 		test.That(t, err, test.ShouldBeNil)
-		test.That(t, holder.ComponentReference, test.ShouldEqual, "mysensor")
+		test.That(t, position.ComponentReference, test.ShouldEqual, "mysensor")
+		positionIsZero(t, position)
 
-		test.That(t, holder.X, test.ShouldEqual, 0)
-		test.That(t, holder.Y, test.ShouldEqual, 0)
-		test.That(t, holder.Z, test.ShouldEqual, 0)
+		// test getPointCloudMap now returns a non empty result
+		pcd, err = vc.getPointCloudMap()
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, pcd, test.ShouldNotBeNil)
+		pc, err := pointcloud.ReadPCD(bytes.NewReader(pcd))
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, pc.Size(), test.ShouldEqual, 1024)
 
-		test.That(t, holder.Imag, test.ShouldEqual, 0)
-		test.That(t, holder.Jmag, test.ShouldEqual, 0)
-		test.That(t, holder.Kmag, test.ShouldEqual, 0)
-		test.That(t, holder.Real, test.ShouldEqual, 1)
+		// third sensor reading populates the pointcloud map and the  position
+		t.Log("sensor reading 3")
+		timestamp = timestamp.Add(time.Second * 2)
+		testAddSensorReading(t, vc, "mysensor", "viam-cartographer/mock_lidar/2.pcd", timestamp, pointcloud.PCDBinary)
 
-		// test getPointCloudMap
-		_, err = vc.getPointCloudMap()
-		test.That(t, err, test.ShouldResemble, errors.New("nil pointcloud"))
+		// test getPosition, is no longer zeroed
+		position, err = vc.getPosition()
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, position.ComponentReference, test.ShouldEqual, "mysensor")
+		test.That(t, position.X, test.ShouldAlmostEqual, -1.4367625864016951, tol)
```

Comment:
These are the same values in the C++ tests: https://github.com/viamrobotics/viam-cartographer/pull/178/files#diff-0bd1cb9b5092e5c3c4e4ffa3be356ac99759fdc0c92546d1d7de11aed037dde0R750

Reply:
I had to remove the precision of these tests as the behavior is different on mac & linux: https://viam.atlassian.net/browse/RSDK-3866